osidb/sync_manager.py

```diff
@@ -0,0 +1,614 @@
+        collector = collectors.FlawCollector()
+        try:
+            collector.sync_flaw(flaw_id)
+        except Exception as e:
+            download.failed(e)
+        else:
+            download.finished()
+        finally:
+            collector.free_queriers()
+
+    def update_synced_links(self):
+        from osidb.models import Flaw
+
+        flaws = Flaw.objects.filter(meta_attr__bz_id=self.sync_id)
+        flaws.update(download_manager=self)
+
+    def __str__(self):
+        from osidb.models import Flaw
+
+        result = super().__str__()
+
+        flaws = Flaw.objects.filter(meta_attr__bz_id=self.sync_id)
+        cves = [f.cve_id for f in flaws]
+        result += f"Flaws: {cves}\n"
+
+        return result
+
+
+class BZTrackerDownloadManager(SyncManager):
+    """
+    Sync manager class for Bugzilla => OSIDB Tracker synchronization.
+    """
+
+    @staticmethod
+    @app.task(name="sync_manager.bz_tracker_download", bind=True)
+    def sync_task(self, tracker_id):
+        from collectors.bzimport import collectors
+
+        download = BZTrackerDownloadManager.get_sync_manager(tracker_id)
+        download.started(self)
+
+        set_user_acls(settings.ALL_GROUPS)
+
+        # Code adapted from collectors.bzimport.collectors.BugzillaTrackerCollector.collect
+        collector = collectors.BugzillaTrackerCollector()
+        try:
+            collector.sync_tracker(tracker_id)
+
+            # Schedule linking tracker => affect
+            link_manager = BZTrackerLinkManager.get_sync_manager(tracker_id)
+            link_manager.schedule()
+        except Exception as e:
+            download.failed(e)
+        else:
+            download.finished()
+        finally:
+            collector.free_queriers()
+
+    def update_synced_links(self):
+        from osidb.models import Tracker
+
+        trackers = Tracker.objects.filter(external_system_id=self.sync_id)
+        trackers.update(download_manager=self)
+
+    def __str__(self):
+        from osidb.models import Tracker
+
+        result = super().__str__()
+
+        trackers = Tracker.objects.filter(external_system_id=self.sync_id)
+        tracker_ids = [t.external_system_id for t in trackers]
+        result += f"Bugzilla trackers: {tracker_ids}\n"
+
+        return result
+
+
+class BZTrackerLinkManager(SyncManager):
+    """
+    Sync manager class for Bugzilla => OSIDB Tracker synchronization where only links between
+    Tracker and Affects are updated.
+    """
+
+    @staticmethod
+    @app.task(name="sync_manager.bz_tracker_link", bind=True)
+    def sync_task(self, tracker_id):
+        from osidb.models import Affect, Flaw, Tracker
+
+        linker = BZTrackerLinkManager.get_sync_manager(tracker_id)
+        linker.started(self)
+
+        set_user_acls(settings.ALL_GROUPS)
+
+        # Code adapted from collectors.bzimport.convertors.BugzillaTrackerConvertor.affects
+        try:
```

Comment:
The containment of this `try` block should be I think moved into a separate method so it can be individually called in tests